### PR TITLE
Update PETSC

### PR DESCRIPTION
**Pull request by akleb**
## Purpose
Update the latest PETSC version to 3.15 (mdolab/docker#139)

## Type of change
What types of change is it?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
Explain the steps needed to test the new code to verify that it does indeed address the issue and produce the expected behavior.

## Checklist

- [X] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [X] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: src/3D/3D_elliptic.F90 [hunk 1]
```diff
@@ -502,8 +503,13 @@ subroutine generateSolution
   call KSPSetTolerances(ellipKSP, kspRelTol, 1e-16, 1e20, kspMaxIts, ierr)
   call EChk(ierr, __FILE__, __LINE__)
 
+#if PETSC_VERSION_GT(3, 14, 0)
```
Comment (nwu63) [inline src/3D/3D_elliptic.F90:506]: I believe the correct version check would be `PETSC_VERSION_GE(3, 15, 0)` since this renaming happened in 3.15

Comment (akleb) [reply src/3D/3D_elliptic.F90:506]: Ah, I just thought we were checking greater than 3.14, but this makes sense

**Review by nwu63 — APPROVED** [review]

**Review by sseraj — APPROVED** [review]
This looks fine to me based on what I know of the PETSc API changes. @nwu63 please merge if the plan is to merge the PETSc PRs individually (rather than all at once)

**Comment by nwu63** [comment]
Because this should not break anything, we plan to merge these in individually, then test them altogether once the deps images are updated to 3.15.

**Pull request merged**